qcengine/programs/orca.py

```diff
@@ -0,0 +1,297 @@
+        #         molpro_version = year + "." + minor
+        #         self.version_cache[which_prog] = safe_version(molpro_version)
+
+        return "4.2.1"
+
+    def compute(self, input_data: "AtomicInput", config: "JobConfig") -> "AtomicResult":
+        """
+        Run Orca
+        """
+        # Check if Orca executable is found
+        self.found(raise_error=True)
+
+        # Check Orca version
+        if parse_version(self.get_version()) < parse_version("4.2.1"):
+            raise TypeError("Orca version '{}' not supported".format(self.get_version()))
+
+        # Setup the job
+        job_inputs = self.build_input(input_data, config)
+
+        # Run Orca
+        exe_success, proc = self.execute(job_inputs)
+
+        # Determine whether the calculation succeeded
+        if exe_success:
+            # If execution succeeded, collect results
+            result = self.parse_output(proc, input_data)
+            return result
+        else:
+            # Return UnknownError for error propagation
+            return UnknownError(proc["stderr"])
+
+    def execute(
+        self,
+        inputs: Dict[str, Any],
+        extra_infiles: Optional[Dict[str, str]] = None,
+        extra_outfiles: Optional[List[str]] = None,
+        as_binary: Optional[List[str]] = None,
+        extra_commands: Optional[List[str]] = None,
+        scratch_name: Optional[str] = None,
+        scratch_messy: bool = False,
+        timeout: Optional[int] = None,
+    ) -> Tuple[bool, Dict[str, Any]]:
+        """
+        For option documentation go look at qcengine/util.execute
+        """
+        infiles = inputs["infiles"]
+        if extra_infiles is not None:
+            infiles.update(extra_infiles)
+
+        # Collect all output files and update with extra_outfiles
+        outfiles = ["dispatch.engrad"]
+
+        if extra_outfiles is not None:
+            outfiles.extend(extra_outfiles)
+
+        # Replace commands with extra_commands if present
+        commands = inputs["commands"]
+        if extra_commands is not None:
+            commands = extra_commands
+
+        # Run the Orca program
+        exe_success, proc = execute(
+            commands,
+            infiles=infiles,
+            outfiles=outfiles,
+            as_binary=as_binary,
+            scratch_name=scratch_name,
+            scratch_directory=inputs["scratch_directory"],
+            scratch_messy=scratch_messy,
+            timeout=timeout,
+        )
+        return exe_success, proc
+
+    def build_input(
+        self, input_model: "AtomicInput", config: "JobConfig", template: Optional[str] = None
+    ) -> Dict[str, Any]:
+
+        if template is None:
+            input_file = []
+
+            # Resolving keywords
+            # caseless_keywords = {k.lower(): v for k, v in input_model.keywords.items()}
+
+            # Write the geom
+            xyz_block = input_model.molecule.to_string(dtype="orca", units="Angstrom")
```

Comment:
there isn't an orca dtype at the moment. you can add one [here](https://github.com/MolSSI/QCElemental/blob/master/qcelemental/molparse/to_string.py#L386) or work with me so I can write one.

also, running in bohr is strongly preferred since atomic units don't change with codata revisions.

Reply:
Along with this PR there is another PR for QCElemental. I will create it soon. I will change to bohr instead. Let me research about it.